pkg/cli/serve/serve.go

```diff
@@ -130,13 +186,54 @@ func (o *serveOptions) Run() error {
 		return err
 	}
 
-	// Initialize ocm sdk connection client
-	sdkclient, err := ocm.NewConnection().Build(viper.GetString(config.OcmURL),
-		viper.GetString(config.ClusterID),
-		viper.GetString(config.AccessToken))
-	if err != nil {
-		o.logger.WithError(err).Fatal("Can't initialise OCM sdk.Connection client")
-		return err
+	// Depending on whether the FleetMode is enabled or not, we need to initiate the OCM SDK connection accordingly
+	// If fleet mode is not enabled, we will fetch the cluster ID and access token to initiate connection with OCM
+	if !o.fleetMode {
+		sdkclient, err = ocm.NewConnection().Build(viper.GetString(config.OcmURL),
+			viper.GetString(config.ClusterID),
+			viper.GetString(config.AccessToken))
+		if err != nil {
+			o.logger.WithError(err).Fatal("Can't initialise OCM sdk.Connection client in non-fleet mode")
+			return err
+		}
+		o.logger.Info("Connection with OCM initialised successfully in non-fleet mode")
+	} else {
+		// If fleet mode is enabled, the connection to OCM needs to initiate using client ID and client secret
+		// On the managed cluster, the client ID and secret will be fetched from the secret volume however for
+		// local testing, the client ID and secret can be passed directly as flags for ocm-agent CLI.
+		ocmAgentClientID = viper.GetString(config.OCMClientID)
+		ocmAgentClientSecret = viper.GetString(config.OCMClientSecret)
+		ocmAgentURL = viper.GetString(config.OcmURL)
+
+		if ocmAgentClientID == "" && ocmAgentClientSecret == "" {
+			clientID, err := os.ReadFile(consts.OCMAgentAccessFleetSecretPathBase + os.Getenv("OCM_AGENT_SECRET_NAME") + "/" + consts.OCMAgentAccessFleetSecretClientKey)
+			ocmAgentClientID = string(clientID)
+			if err != nil {
+				o.logger.WithError(err).Fatal("Can't find value for secret key ", consts.OCMAgentAccessFleetSecretClientKey)
+				return err
+			}
+
+			clientSecret, err := os.ReadFile(consts.OCMAgentAccessFleetSecretPathBase + os.Getenv("OCM_AGENT_SECRET_NAME") + "/" + consts.OCMAgentAccessFleetSecretClientSecretKey)
+			ocmAgentClientSecret = string(clientSecret)
+			if err != nil {
+				o.logger.WithError(err).Fatal("Can't find value for secret key ", consts.OCMAgentAccessFleetSecretClientSecretKey)
+				return err
+			}
+
+			url, err := os.ReadFile(consts.OCMAgentAccessFleetSecretPathBase + os.Getenv("OCM_AGENT_SECRET_NAME") + "/" + consts.OCMAgentAccessFleetSecretURLKey)
+			ocmAgentURL = string(url)
+			if err != nil {
+				o.logger.WithError(err).Fatal("Can't find value for secret key ", consts.OCMAgentAccessFleetSecretURLKey)
+				return err
+			}
+		}
+
+		sdkclient, err = sdk.NewConnectionBuilder().URL(ocmAgentURL).Client(ocmAgentClientID, ocmAgentClientSecret).Insecure(false).Build()
```

Comment:
why do we need to initialized the connection here?
we should have the metrics set in the separated webhook receivers. So if the connection with OCM failed, we can know when the webhook receiver call failed as well. And we can also choose to be alerted.

Reply:
Over here as well, the connection to OCM is not actually created. The required structs are referred from OCM SDK to create connection being built and constructed which will depend on the OIDC client ID and secret being passed in the fleet mode.